bootstrap/bootstrap.py

```diff
@@ -334,8 +353,13 @@ def main() -> None:
                         help='bootstrap dependency file')
     parser.add_argument('-w', '--with-compiler', type=Path,
                         help='path to GHC')
+    parser.add_argument('--no-network', action='store_true',
+                        help='disable automatic downloading of source files')
     args = parser.parse_args()
 
+    global network_allowed
```

Comment:
Please, if we manage to thread every other argument, we can do without `global`.

Reply:
Eliminated `global`.